aloha-h2o/src/main/scala/com/eharmony/aloha/models/h2o/H2oModel.scala

```diff
@@ -0,0 +1,204 @@
+package com.eharmony.aloha.models.h2o
+
+import com.eharmony.aloha.factory.{ModelParser, ModelParserWithSemantics, ParserProviderCompanion}
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.io.AlohaReadable
+import com.eharmony.aloha.io.sources.ModelSource
+import com.eharmony.aloha.models.BaseModel
+import com.eharmony.aloha.models.h2o.H2oModel.Features
+import com.eharmony.aloha.models.h2o.categories._
+import com.eharmony.aloha.models.h2o.compiler.Compiler
+import com.eharmony.aloha.models.h2o.json.{H2oSpec, H2oAst, H2oModelJson}
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.score.Scores.Score
+import com.eharmony.aloha.score.basic.ModelOutput
+import com.eharmony.aloha.score.conversions.ScoreConverter
+import com.eharmony.aloha.semantics.Semantics
+import com.eharmony.aloha.semantics.func.GenAggFunc
+import com.eharmony.aloha.util.{EitherHelpers, Logging}
+import hex.genmodel.GenModel
+import hex.genmodel.easy.exception.PredictUnknownCategoricalLevelException
+import hex.genmodel.easy.{EasyPredictModelWrapper, RowData}
+import spray.json.{DeserializationException, JsValue, JsonReader}
+
+import scala.annotation.tailrec
+import scala.collection.{immutable => sci}
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Created by deak on 9/30/15.
+ */
+// TODO: Need to make sure the model source is used rather than a model.
+//       This is because we want to protected against a user compiling a model locally and serializing it.
+//       Then deserializing on a cluster where the class files don't exist.  If we follow the same methodology as
+//       the VW model, then we can get the same guarantees.
+final case class H2oModel[-A, +B](
+    modelId: ModelIdentity,
+    modelSource: ModelSource,
+    featureNames: sci.IndexedSeq[String],
+    featureFunctions: sci.IndexedSeq[FeatureFunction[A]],
+    numMissingThreshold: Option[Int] = None)(implicit private[this] val scb: ScoreConverter[B])
+  extends BaseModel[A, B]
+     with Logging {
+
+  // Because H2o's RowData object is essentially a Map of String to Object, we unapply the wrapper
+  // and throw away the type information on the function return type.  We have type safety because
+  // FeatureFunction is sealed (ADT).
+  @transient private[this] lazy val lazyAnyRefFF = featureFunctions map {
```

Comment:
I hate having to throw away the typed information here but I don't know whittles to do.